zipline/finance/trading.py

```diff
@@ -486,7 +456,8 @@ class SimulationParameters(object):
     def __init__(self, period_start, period_end,
                  capital_base=10e3,
                  emission_rate='daily',
-                 data_frequency='daily'):
+                 data_frequency='daily',
+                 env=None):
```

Comment:
SimulationParameters accept an optional arg of an env that allows the SimulationParameters to initialize the env-dependent fields. If no env is passed, those fields can be initialized later by using the 'update_internal_from_env(env)' method.